tasty-query/shared/src/main/scala/tastyquery/Symbols.scala

```diff
@@ -536,9 +470,49 @@ object Symbols {
     private[tastyquery] final def specialErasure(using Context): Option[() => ErasedTypeRef] =
       mySpecialErasure
 
-    protected[this] final def ensureDeclsInitialized()(using Context): Unit =
-      // ClassSymbols are always initialized when created
-      ()
+    // DeclaringSymbol implementation
+
+    private[Symbols] final def addDecl(decl: Symbol): Unit =
+      val set = myDeclarations.getOrElseUpdate(decl.name, new mutable.HashSet)
+      if decl.isType then assert(set.isEmpty, s"trying to add a second entry $decl for type name ${decl.name} in $this")
+      set += decl
+
+    private[Symbols] final def hasOverloads(name: SignedName): Boolean =
+      myDeclarations.get(name.underlying) match
+        case Some(decls) => decls.sizeIs > 1
+        case _           => false
+
+    final def getDecls(name: Name)(using Context): List[Symbol] =
+      name match
+        case name: SignedName => getDecl(name).toList
+        case _                => myDeclarations.get(name).fold(Nil)(_.toList)
+
+    final def getDecl(name: Name)(using Context): Option[Symbol] =
+      name match
+        case overloaded: SignedName =>
+          distinguishOverloaded(overloaded)
+        case name =>
+          myDeclarations.get(name) match
+            case Some(decls) =>
+              if decls.sizeIs == 1 then Some(decls.head)
+              else if decls.sizeIs > 1 then
+                throw MemberNotFoundException(this, name, s"unexpected overloads in $this: ${decls.mkString(", ")}")
```

Comment:
This is not compatible with a class that has overloads without parameters:
e.g.
```scala
class Foo:
  def foo = 23
  def foo(i: Int) = i
```

Reply:
That logic was already there before:
https://github.com/scalacenter/tasty-query/pull/162/files#diff-5977ded6850927706fcf3030a7eccf82551476d77a8bb2e7d962cae7b91bb81dL401-L405

We should definitely fix it, but I don't think it belongs in this PR.